docs/articles/vector_search_based_recommender_system.md

```diff
@@ -0,0 +1,276 @@
+
+## Introduction
+A real-time recommender system with an optimized and efficient [architecture](https://rockset.com/blog/a-blueprint-for-a-real-world-recommendation-system/) can add value to an organization by enhancing personalization, user engagement, and ultimately increasing user satisfaction.
+
+Building a recommendation system that efficiently deals with high-dimensional data to find accurate, relevant, and similar items in a large dataset requires effective and efficient vectorization, vector indexing, search, and retrieval. This, in turn, demands robust databases with vector capabilities. For this post, we will use Rockset as the database and OpenAI embedding models to vectorize the dataset.  Other vector databases are available, you can find examples [here](https://vdbs.superlinked.com/).
+
+There are several components of vector search required to build a recommendation engine, we won't dive into them here however you can read more about them [here](https://superlinked.com/vectorhub/building-blocks/vector-search/introduction). The key pieces for this example are a database optimized for metadata filtering, vector search, and keyword search, supporting sub-second search, aggregations, and joins at scale, is needed.
+
+**Overview of the Recommendation WebApp**
+The image below shows the workflow of the application we'll be building. We have unstructured data i.e., game reviews in our case. We'll generate vector embeddings for all of these reviews through OpenAI model and store them in the database. Then we'll use the same OpenAI model to generate vector embeddings for our search query and match it with the review vector embeddings using a similarity function such as the nearest neighbor search, dot product or approximate neighbor search. Finally, we will have our top 10 recommendations ready to be displayed.
+
+![overview](../assets/use_cases/vector_search_based_recommender_system/image3.png)
+
+## Steps to build the Recommender System using Rockset and OpenAI Embedding
+
+Let's begin with signing up for Rockset and OpenAI and then dive into all the steps involved within the Google Colab notebook to build our recommendation webapp:
+
+### Step 1: Initiate your vector database
+
+In this example we are using [Rockset](https://rockset.com/create/) as a vector database. Once signed up, create an [API key](https://console.rockset.com/apikeys) to use in the backend code:
+
+```python
+import os
+os.environ["ROCKSET_API_KEY"] = "XveaN8L9mUFgaOkffpv6tX6VSPHz####"
+```
+
+### Step 2: Create a new Collection and Upload Data
+
+For this tutorial, we'll be using Amazon product review [data](https://drive.google.com/file/d/1EEvCUqKIH6LuLLGRjo6ayCEXqT4Z-ZVE/view?usp=drive_link). Download this on your local machine so it can be uploaded to your collection.
+
```

Comment:
this Amazon product review data = the "game reviews" you refer to above?

Reply:
yup, it's all the same data